log_test.go

```diff
@@ -78,6 +81,57 @@ func TestInfo(t *testing.T) {
 	})
 }
 
+func TestFatal(t *testing.T) {
+	t.Run("should exit", func(t *testing.T) {
+		if os.Getenv("TEST_FATAL") == "1" {
+			log := New(os.Stderr)
+			log.Fatal().Msg("")
+			return
+		}
+
+		cmd := exec.Command(os.Args[0], "-test.run=TestFatal/should_exit")
+		cmd.Env = append(os.Environ(), "TEST_FATAL=1")
+		stderr, err := cmd.StderrPipe()
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = cmd.Start()
+		if err != nil {
+			t.Fatal(err)
+		}
+		out, err := io.ReadAll(stderr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = cmd.Wait()
+		if err == nil {
+			t.Error("Expected log Fatal to exit with non-zero status")
+		}
+
+		if got, want := decodeIfBinaryToString(out), `{"level":"fatal"}`+"\n"; got != want {
+			t.Errorf("invalid log output:\n got:  %v\nwant: %v", got, want)
+		}
+	})
+
+	t.Run("should not exit", func(t *testing.T) {
+		out := &bytes.Buffer{}
+		log := New(out).ExitFunc(func(_ int) {})
+		log.Fatal().Msg("")
+		if got, want := decodeIfBinaryToString(out.Bytes()), `{"level":"fatal"}`+"\n"; got != want {
+			t.Errorf("invalid log output:\n got:  %v\nwant: %v", got, want)
+		}
```

Comment:
Adding such a thing would help to validate the exiter is called

```suggestion
		out := &bytes.Buffer{}
		var receivedCode int
		log := New(out).ExitFunc(func(code int) {
			receivedCode = code
		})
		log.Fatal().Msg("")
		if got, want := receivedCode, 1; got != want {
			t.Errorf("invalid exitCode:\n got:  %v\nwant: %v", got, want)
		}
		if got, want := decodeIfBinaryToString(out.Bytes()), `{"level":"fatal"}`+"\n"; got != want {
			t.Errorf("invalid log output:\n got:  %v\nwant: %v", got, want)
		}
```